### ISSUE TITLE
Add access control for upstream backends

### ISSUE THREAD

**Issue by vwuerbel** [avenga/couper#8]

**Comment by filex** [comment]
spec proposal:

```
backend "foo" {
  origin = "…"
  basic_auth = "user:pass"
}
```

`user:pass` is then base64 encoded and send as `Authorization` header with `Basic ` prefix.

**Comment by filex** [comment]
current workaround: set the header manually:

```hcl
server "example" {
    api {
        endpoint "/**" {
            backend {
                origin = "https://httpbin.org"
                request_headers = {
                    authorization = "Basic ${base64_encode("foo:bar")}"
                }
            }
        }
    }
}
```

**Comment by filex** [comment]
`basic_auth` was implemented in 91d287fd
